php_phongo.c

```diff
@@ -972,60 +1015,93 @@ static bool php_phongo_apply_options_to_uri(mongoc_uri_t *uri, bson_t *options T
 			continue;
 		}
 
-		if (mongoc_uri_option_is_utf8(key) && BSON_ITER_HOLDS_UTF8(&iter)) {
+		if (mongoc_uri_option_is_utf8(key)) {
+			if (!BSON_ITER_HOLDS_UTF8(&iter)) {
+				PHONGO_URI_INVALID_TYPE(iter, "string");
+				return false;
+			}
+
 			if (!mongoc_uri_set_option_as_utf8(uri, key, bson_iter_utf8(&iter, NULL))) {
-				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Failed to parse \"%s\" URI option", key);
+				/* Assignment uses mongoc_uri_set_appname() for the "appname"
+				 * option, which validates length in addition to UTF-8 encoding.
+				 * For BC, we report the invalid string to the user. */
```

Comment:
AFAICT, "appname" was only being handled in `phongo_manager_init()` because `mongoc_uri_set_appname()` also checked the length in addition to basic UTF-8 validation and we wanted to use a more specific error message than "Failed to parse". Since `mongoc_uri_option_is_utf8()` does consider "appname" a string option and libmongoc supports assignment with `mongoc_uri_set_option_as_utf8()`, we can handle the special error reporting here and reduce some clutter in `phongo_manager_init()`.